### PR TITLE
Move the path simplification library into its own package

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @tehcyx 
/assign @Katharine @cjwagner

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
WTF?

```
 @@ -69,6 +69,7 @@ require (
 	k8s.io/code-generator v0.0.0-20190831074504-732c9ca86353
 	k8s.io/klog v0.4.0
 	k8s.io/repo-infra v0.0.0-20190806171332-9f4571ad7242
+	k8s.io/utils v0.0.0-20190506122338-8fab8cb257d5
 	mvdan.cc/xurls/v2 v2.0.0
 	sigs.k8s.io/controller-runtime v0.2.1
 	sigs.k8s.io/yaml v1.1.0
ERROR: modules changed. Update with ./hack/update-deps.sh 
```

/retets

File: prow/simplifypath/simplify_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package simplifypath
+
+import (
+	"testing"
+
+	"k8s.io/utils/diff"
```
Comment (fejta) [inline prow/simplifypath/simplify_test.go]: Because of this

Comment (stevekuznetsov) [reply prow/simplifypath/simplify_test.go]: huh, apparently that is different from "k8s.io/apimachinery/pkg/util/diff"

**Comment by stevekuznetsov** [comment]
@cjwagner @Katharine @fejta this is ready now, I will do a migration to `k8s.io/utils` out-of-band

**Pull request merged**